### PR TITLE
feat: ✨ add `create_root_path()`

### LINKED ISSUE #681: Create and test`get_root_envvar()` - a part of `path_sprout_root()`
Signature: `create_package_structure() -> Path | None:`

Returns: 
- Path if the `SPROUT_ROOT` environment variable is set
- None if `SPROUT_ROOT` isn't set

Context: see the `path_sprout_root()`diagram in the [path functions section](https://sprout.seedcase-project.org/docs/design/implementation/python-functions#path-functions) in the docs.

### LINKED ISSUE #682: Create and test `create_root_path()` - a part of `path_sprout_root()`
Signature: `create_root_path() -> Path:`

Returns: Path that ends with `sprout/`

Context: 
- This function is used if the envvar `SPROUT_ROOT` hasn't been set and `get_root_envvar()` returns `None` 
- It uses the `platformdirs` package internally 
- See the `path_sprout_root()`diagram in the [path functions section](https://sprout.seedcase-project.org/docs/design/implementation/python-functions#path-functions) in the docs.

### PR DESCRIPTION
**Pull request by signekb**
## Description

Closes #682 

## Reviewer Focus

This PR needs an in-depth review.

## Checklist

- [X] Added or updated tests
- [X] Tests passed locally
- [X] Linted and formatted code
- [X] Updated documentation (=docstrings)

### CODE REVIEW

File: sprout/core/create_root_path.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from platformdirs import user_data_path
+
+
+def create_root_path() -> Path:
+    """Creates the path to Sprout root.
+
+    Returns:
+        A Path with the Sprout root directory.
+    """
+    root_path = user_data_path("sprout")
```
Comment (signekb) [inline sprout/core/create_root_path.py:12]: If we want it to be created if it doesn't exist, we can add the argument [`ensure_exists=True`](https://platformdirs.readthedocs.io/en/latest/api.html#platformdirs.api.PlatformDirsABC.ensure_exists)

File: tests/core/test_create_root_path.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from sprout.core.create_root_path import create_root_path
```
Comment (signekb) [inline tests/core/test_create_root_path.py:1]: Am I missing some test cases? 
E.g., would we expect that it already exist at this point?  I.e., `.exists() is True`?

File: sprout/core/create_root_path.py (hunk 1 above)
Comment (signekb) [inline sprout/core/create_root_path.py:12]: This doesn't create the directory; just the path

File: tests/core/test_create_root_path.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+
+def test_returns_root_path():
+    """Returns root path o."""
```
Comment (lwjohnst86) [inline tests/core/test_create_root_path.py:5]:
```suggestion
    """Returns root path."""
```

?

File: tests/core/test_create_root_path.py (hunk 2 above)
Comment (lwjohnst86) [reply tests/core/test_create_root_path.py:1]: I think this is fine for now ☺️ One of the main purposes of tests are also to just make sure it runs! 😛

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Looks great! Only a very minor change.

File: sprout/core/create_root_path.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+def create_root_path() -> Path:
+    """Creates the path to Sprout root.
+
+    Returns:
+        A Path with the Sprout root directory tied to the user.
+    """
+    root_path = user_data_path("sprout")
+
+    return root_path
```
Comment (martonvago) [inline sprout/core/create_root_path.py:14]:
```suggestion
    return user_data_path("sprout")
```
Could just return this directly

**Review by martonvago — CHANGES_REQUESTED** [review]
Nice 👍 👍 !!

File: sprout/core/create_root_path.py (hunk 4 above)
Comment (signekb) [reply sprout/core/create_root_path.py:14]: You're right! ⭐

**Review by martonvago — APPROVED** [review]

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**